app/src/main/java/ch/hikemate/app/ui/map/HikeDetailScreen.kt

```diff
@@ -102,9 +101,15 @@ fun HikeDetailScreen(
 
   val context = LocalContext.current
 
+  LaunchedEffect(listOfHikeRoutesViewModel.selectedHikeRoute.collectAsState().value) {
+    if (listOfHikeRoutesViewModel.selectedHikeRoute.value == null) {
+      navigationActions.goBack()
+    }
+  }
+
   if (listOfHikeRoutesViewModel.selectedHikeRoute.collectAsState().value == null) {
-    Toast.makeText(context, "No route selected, returning to map.", Toast.LENGTH_SHORT).show()
-    navigationActions.goBack()
+    Log.e("HikeDetailScreen", "No selected hike route")
```

Comment:
Question: What is the point of this `if`-clause, if we basically assert that its not null via the launched effect?

Reply:
LaunchedEffect isn't guaranteed to run before we got to the not-null assertion below